container/lxd/initialisation.go

```diff
@@ -49,6 +76,147 @@ func getPackagingConfigurer(series string) (config.PackagingConfigurer, error) {
 	return config.NewPackagingConfigurer(series)
 }
 
+func configureZFS() {
+	/* create a 100 GB pool by default (sparse, so it won't actually fill
+	 * that immediately)
+	*/
+	output, err := exec.Command(
+		"lxd",
+		"init",
+		"--auto",
+		"--storage-backend", "zfs",
+		"--storage-pool", "lxd",
+		"--storage-create-loop", "100",
+	).CombinedOutput()
+
+	if err != nil {
+		logger.Warningf("configuring zfs failed with %s: %s", err, string(output))
+	}
+}
+
+func configureLXDBridge() error {
+	f, err := os.OpenFile(lxdBridgeFile, os.O_RDWR, 0777)
+	if err != nil {
+		/* We're using an old version of LXD which doesn't have
+		 * lxd-bridge; let's not fail here.
+		 */
+		if os.IsNotExist(err) {
+			logger.Warningf("couldn't find %s, not configuring it", lxdBridgeFile)
+			return nil
+		}
+		return errors.Trace(err)
+	}
+	defer f.Close()
+
+	output, err := exec.Command("ip", "addr", "show").CombinedOutput()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	subnet, err := detectSubnet(string(output))
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	existing, err := ioutil.ReadAll(f)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	result := editLXDBridgeFile(string(existing), subnet)
+	_, err = f.Seek(0, 0)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	_, err = f.WriteString(result)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+
+	/* non-systemd systems don't have the lxd-bridge service, so this always fails */
+	_ = exec.Command("service", "lxd-bridge", "restart").Run()
+	return exec.Command("service", "lxd", "restart").Run()
+}
+
+func detectSubnet(ipAddrOutput string) (string, error) {
```

Comment:
On Mon, Mar 28, 2016 at 11:20:48PM -0700, John Arbash Meinel wrote:
> > +	if err != nil {
> > +		return errors.Trace(err)
> > +	}
> > +
> > +	_, err = f.WriteString(result)
> > +	if err != nil {
> > +		return errors.Trace(err)
> > +	}
> > +
> > +
> > +	/* non-systemd systems don't have the lxd-bridge service, so this always fails */
> > +	_ = exec.Command("service", "lxd-bridge", "restart").Run()
> > +	return exec.Command("service", "lxd", "restart").Run()
> > +}
> > +
> > +func detectSubnet(ipAddrOutput string) (string, error) {
> 
> I feel like we're missing a fair amount of direct testing of this function

What do you mean by "direct" testing?

> , since we're dealing with "external" data we need a fair amount of confidence that we can handle weird network cases sanely.
> 
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/juju/juju/pull/4903/files/af7d4537ba59659bb1087fc418f257280034e6f2#r57676540